pkg/addons/ensure.go

```diff
@@ -140,6 +141,32 @@ func collectAddons(s *state.State) []addonAction {
 		addonsToDeploy = ensureCCMAddons(s, addonsToDeploy)
 	}
 
+	var foundDefaultStorageClass bool
+
+	if s.Cluster.Addons != nil && !slices.ContainsFunc(s.Cluster.Addons.Addons, func(addon kubeoneapi.Addon) bool {
+		return addon.Name == resources.AddonDefaultStorageClass
+	}) {
+		var existingStorageClasses storagev1.StorageClassList
+
+		if s.DynamicClient != nil {
+			if err := s.DynamicClient.List(s.Context, &existingStorageClasses); err != nil {
+				s.Logger.Error(fail.KubeClient(err, "listing existing storage classes"))
+			}
+		}
+
+		for _, sc := range existingStorageClasses.Items {
+			if sc.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
+				foundDefaultStorageClass = true
+			}
+		}
+	}
+
+	if !foundDefaultStorageClass {
+		addonsToDeploy = append(addonsToDeploy, addonAction{
+			name: resources.AddonDefaultStorageClass,
+		})
+	}
```

Comment:
We should have a way to disable deployment of this addon. Do we already have that in place?

Reply:
no. there was no such way.

Reply:
Well, with delete: true it will be deleted everytime (which after the first time will have no affect any more). And once the used will have own default SC, this addon will be stopped from adding.